central/declarativeconfig/manager_impl.go

```diff
@@ -151,14 +190,97 @@ func (m *managerImpl) runReconciliation() {
 }
 
 func (m *managerImpl) reconcileTransformedMessages(transformedMessagesByHandler map[string]protoMessagesByType) {
```

Comment:
We should definitely add tests for upserting.

Reply:
Do you mean datastore tests?

Reply:
I guess both? It'd be nice to test the flow once, specifically a) the `reconcileTransformedMessages` one regarding transforming messages and calling the updaters (with mocked updaters), ensuring we handle errors correctly as well as call the expected updaters and b) yeah datastore tests as well (seems like for role, permission set, access scope we already have some, but would be great to have one for group as well).